### DISCUSSION TITLE
Merge and rebase strategy

### DISCUSSION THREAD

**Discussion by baparham** [dartclub/turf_dart#28] [General]
[enter opinionated territory 😉]
Hey @tobrun and @lukas-h I was just glancing at the git history, and the merge commits are getting messy. I am personally a fan of using "rebase and merge" strategies in most cases, and falling back to simple merge commits for longer running feature branches that have many commits.

What do you guys think of defaulting to Rebase and Merge as the normal case, and using normal Merge Commits in the use case for example going forward?

**Comment by lukas-h** [comment]
Yes, sorry the git history from yesterday is not very clean because of all the stuff I did with GitHub Actions..
I agree, "rebase and merge" together with PRs on GitHub should be the strategy!

**Comment by tobrun** [comment]
Rebase/Squash and merge is :+1: fwiw even disable merge commits fully in my own repositories. Even for long running feature branches that should work, I try to periodically to rebase when developing independently and if new changes land in between.

**Comment by baparham** [comment]
I updated the repo settings to support rebase and squash, and disabled merge commits for now.
